codex-rs/core/src/windows.rs

```diff
@@ -0,0 +1,250 @@
+    *WSL_AVAILABLE.get_or_init(|| {
+        StdCommand::new("wsl")
+            .arg("--status")
+            .status()
+            .map(|status| status.success())
+            .unwrap_or(false)
+    })
+}
+
+#[cfg(target_os = "windows")]
+fn maybe_print_notice() {
+    if hide_wsl_notice() {
+        return;
+    }
+    if !NOTICE_PRINTED.swap(true, Ordering::SeqCst) {
+        eprintln!(
+            "Detected prefer_wsl=true; relaunching command inside WSL for sandboxed execution."
+        );
+    }
+}
+
+#[cfg(target_os = "windows")]
+fn sandbox_overrides_for_wsl(sandbox_policy: &SandboxPolicy) -> io::Result<Vec<String>> {
+    let mut overrides = Vec::new();
+    match sandbox_policy {
+        SandboxPolicy::DangerFullAccess => {
+            overrides.push("sandbox_mode=\"danger-full-access\"".to_string());
+        }
+        SandboxPolicy::ReadOnly => {
+            overrides.push("sandbox_mode=\"read-only\"".to_string());
+        }
+        SandboxPolicy::WorkspaceWrite {
+            writable_roots,
+            network_access,
+            exclude_tmpdir_env_var,
+            exclude_slash_tmp,
+        } => {
+            overrides.push("sandbox_mode=\"workspace-write\"".to_string());
+            let mut workspace = SandboxWorkspaceWrite {
+                writable_roots: Vec::new(),
+                network_access: *network_access,
+                exclude_tmpdir_env_var: *exclude_tmpdir_env_var,
+                exclude_slash_tmp: *exclude_slash_tmp,
+            };
+            for root in writable_roots {
+                if let Some(converted) = convert_path_to_wsl(root) {
+                    workspace.writable_roots.push(PathBuf::from(converted));
+                }
+            }
+            let json = serde_json::to_string(&workspace).map_err(|err| {
+                io::Error::new(
+                    io::ErrorKind::InvalidData,
+                    format!("failed to serialize sandbox workspace: {err}"),
+                )
+            })?;
+            overrides.push(format!("sandbox_workspace_write={json}"));
+        }
+    }
+    Ok(overrides)
+}
+
+#[cfg(target_os = "windows")]
+fn convert_path_to_wsl(path: &Path) -> Option<String> {
+    let absolute = if path.is_absolute() {
+        path.to_path_buf()
+    } else {
+        std::env::current_dir().map(|cwd| cwd.join(path)).ok()?
+    };
+    let path_str = absolute.to_string_lossy().to_string();
+    run_wslpath(&path_str)
+        .or_else(|| drive_path_to_wsl(&path_str))
+        .map(|converted| converted.trim().to_string())
+}
+
+#[cfg(target_os = "windows")]
+fn run_wslpath(path: &str) -> Option<String> {
+    let output = StdCommand::new("wsl")
+        .arg("wslpath")
+        .arg("-a")
+        .arg(path)
+        .output()
+        .ok()?;
+    if !output.status.success() {
+        return None;
+    }
+    String::from_utf8(output.stdout).ok()
+}
+
+#[cfg(target_os = "windows")]
+fn drive_path_to_wsl(path: &str) -> Option<String> {
```

Comment:
Please do not hand-roll this sort of thing. Look at https://crates.io/crates/dunce or see what else is out there.